### PR TITLE
fix #55489

### LINKED ISSUE #55489: BUG: pandas.read_excel() with dtype=object is broken in pandas 2.1.1
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

Create an excel file nammed `xl_file.xlsx`, with one worksheet named ``Sheet1`` as following:

|    A    |      B      |   C    |
|---------|-------------|--------|
|    2    | 'dummy_str  |        |

Then save it and run this code in the directory where you created the excel file :

```python
import pandas as pd
from pathlib import Path

df = pd.read_excel(
    io=Path("xl_file.xlsx"),
    sheet_name="Sheet1",
    dtype=object,
)

dtypes_1 = df.dtypes
dtypes_2 = df.fillna("").dtypes
```

``` python
>>> print(dtypes_1)
0   A    object # <- expected int, or int64
1   B    object
2   C    object
```

``` python
>>> print(dtypes_2)
0   A     int64 # <- when we use df.fillna(""), we have the expected behavior
1   B    object
2   C    object
```

### Issue Description

The method ``pandas.read_excel()`` with argument ``dtypes=object`` does not prevent pandas from converting data types, as described on the [offical documentation page](https://pandas.pydata.org/docs/reference/api/pandas.read_excel.html)

### Expected Behavior

when we use the method ``pandas.read_excel()`` with argument ``dtypes=object``, on an Excel workbook which contains int values, the dtypes of this columns should be ``int64``

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit              : e86ed377639948c64c429059127bcf5b359ab6be
python              : 3.11.5.final.0
python-bits         : 64
OS                  : Windows
OS-release          : 10
Version             : 10.0.19044
machine             : AMD64
processor           : Intel64 Family 6 Model 141 Stepping 1, GenuineIntel
byteorder           : little
LC_ALL              : None
LANG                : fr_FR.UTF-8
LOCALE              : fr_FR.cp1252

pandas              : 2.1.1
numpy               : 1.24.2
pytz                : 2023.3
dateutil            : 2.8.2
setuptools          : 65.5.0
pip                 : 23.2.1
Cython              : None
pytest              : 7.3.1
hypothesis          : None
sphinx              : None
blosc               : None
feather             : None
xlsxwriter          : 3.1.7
lxml.etree          : None
html5lib            : None
pymysql             : None
psycopg2            : None
jinja2              : 3.1.2
IPython             : None
pandas_datareader   : None
bs4                 : None
bottleneck          : None
dataframe-api-compat: None
fastparquet         : None
fsspec              : None
gcsfs               : None
matplotlib          : None
numba               : None
numexpr             : None
odfpy               : None
openpyxl            : 3.1.2
pandas_gbq          : None
pyarrow             : None
pyreadstat          : None
pyxlsb              : None
s3fs                : None
scipy               : None
sqlalchemy          : None
tables              : None
tabulate            : None
xarray              : None
xlrd                : None
zstandard           : None
tzdata              : 2023.3
qtpy                : None
pyqt5               : None

</details>

### PR DESCRIPTION
**Pull request by ziadk**
- [BUG ] closes #55489
- Added _**"This will necessarily result in ``object`` dtype."**_ to the pandas.io.read_excel() docstring

### CODE REVIEW

**Comment by rhshadrach** [comment]
@ziadk - you do not need to update your PR branch each time the main branch is updated. Doing so kicks off all the tests. Periodically (every few days or so) is okay, and especially when there are merge conflicts.

File: pandas/io/excel/_base.py [hunk 1]
```diff
@@ -157,7 +157,7 @@
     Data type for data or columns. E.g. {{'a': np.float64, 'b': np.int32}}
     Use `object` to preserve data as stored in Excel and not interpret dtype.
     If converters are specified, they will be applied INSTEAD
-    of dtype conversion.
+    of dtype conversion.This will necessarily result in ``object`` dtype.
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:160]: Where this currently is, it appears to apply to the previous sentence about dtype conversion. It needs to be moved one sentence prior. Also, I think it'd be more clear do have it read:

> Use `object` to preserve data as stored in Excel and not interpret dtype, which will necessarily result in ``object`` dtype.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: pandas/io/excel/_base.py (hunk 1 above)
Comment (ziadk) [reply pandas/io/excel/_base.py:160]: Thanks a lot ! This is my first contribution so I am getting the hang of it !

**Comment by rhshadrach** [comment]
@ziadk - this looked pretty good, likely just needed to fix the line being too long (we don't allow more than 88 characters on a line) and then add the missing space between the period and the next sentence. Are you interested in continuing?

**Pull request closed without merging**